### PR TITLE
241

### PR DESCRIPTION
**Pull request by SidMalladi**
#241 - Add new functionalites to Phone module @cieslarmichal

### CODE REVIEW

File: include/faker-cxx/Phone.h [hunk 1]
```diff
@@ -25,6 +25,10 @@ class Phone
      * @endcode
      */
     static std::string number(std::optional<std::string> = std::nullopt);
+    static std::string platform();
+    static std::string modelName();
+    static std::string manufacturer();
+
```
Comment (cieslarmichal) [inline include/faker-cxx/Phone.h:31]: please add tests for new methods

File: src/modules/phone/Phone.cpp [hunk 2]
```diff
@@ -42,6 +42,24 @@ std::string Phone::imei()
...
+std::string Phone::modelName()
+{
+    std::vector<std::string> modelNames = {"Samsung Galaxy S9", "iPhone X", "Google Pixel 4"};
+    return Helper::arrayElement(modelNames);
+}
+
+std::string Phone::manufacturer()
+{
+    std::vector<std::string> manufacturers = {"Samsung", "Apple", "Google"};
+    return Helper::arrayElement(manufacturers);
```
Comment (cieslarmichal) [inline src/modules/phone/Phone.cpp:60]: please add more data

**Review by cieslarmichal — CHANGES_REQUESTED** [review]

File: include/faker-cxx/Phone.h (hunk 1 above)
Comment (SidMalladi) [reply include/faker-cxx/Phone.h:31]: Will I be creating a file (Phone.test.cpp?) with the following code:
#include <faker-cxx/Phone.h>
#include <cassert>
#include <iostream>

void testPlatform() {
    std::string result = faker::Phone::platform();
    assert(!result.empty()); // Check that result is not empty
    // Additional checks can be made if there are specific expected platforms
}

void testModelName() {
    std::string result = faker::Phone::modelName();
    assert(!result.empty()); // Check that result is not empty
    // Additional checks can be made if there are specific expected model names
}

void testManufacturer() {
    std::string result = faker::Phone::manufacturer();
    assert(!result.empty()); // Check that result is not empty
    // Additional checks can be made if there are specific expected manufacturers
}

int main() {
    testPlatform();
    testModelName();
    testManufacturer();
    std::cout << "All Phone tests passed.\n";
    return 0;
}

Or do you want it to be done someway specifically?

Comment (cieslarmichal) [reply include/faker-cxx/Phone.h:31]: there is already a test file:
https://github.com/cieslarmichal/faker-cxx/blob/main/src/modules/phone/PhoneTest.cpp

just append new tests in the similar fashion :)

Comment (cieslarmichal) [reply include/faker-cxx/Phone.h:31]: do asserts like:
    ASSERT_TRUE(std::ranges::any_of(platforms, [generatedPlatform](const std::string& platform) { return platform == generatedPlatform; }));

Comment (cieslarmichal) [reply include/faker-cxx/Phone.h:31]: there is a lot of tests file like that in repository, so you have a references

**Comment by SidMalladi** [comment]
Hi @cieslarmichal , I've made the suggested changes, can you please review and merge?

**Review by cieslarmichal — APPROVED** [review]

File: src/modules/phone/PhoneTest.cpp [hunk 3]
```diff
@@ -66,3 +68,38 @@ TEST_F(PhoneTest, NumberFormatTest)
     EXPECT_FALSE(phoneNumber.empty());
     ASSERT_TRUE(isStringNumericWithSpecialChars(phoneNumber));
 }
+
+TEST_F(PhoneTest, PlatformGeneration)
+{
+    std::string generatedPlatform = Phone::platform();
+    std::vector<std::string> platforms = {
```
Comment (cieslarmichal) [inline src/modules/phone/PhoneTest.cpp:75]: missing data in tests, those are not the same vector as in cpp file, extract it to data/ directory and use it in both files please

**Review by cieslarmichal — CHANGES_REQUESTED** [review]
extract data to data/ directory and use it in both cpp and test.cpp files

File: src/modules/phone/PhoneTest.cpp (hunk 3 above)
Comment (SidMalladi) [reply src/modules/phone/PhoneTest.cpp:75]: am I supposed to create a directory file under the data sub-folder? Under the data sub-folder, there only exists an 'PhoneNumbers.h' file.

**Comment by SidMalladi** [comment]
hi @cieslarmichal just made the changes

**Comment by cieslarmichal** [comment]
compilation error -  it should be data/, not faker-cxx/data/

**Comment by cieslarmichal** [comment]
Also, you defined this data in separate file - good, but you are not using it anywhere, should be used instead defining those vectors again in implementation an test cases

**Review by cieslarmichal — APPROVED** [review]

**Pull request merged**